falcon/util/uri.py

```diff
@@ -390,14 +390,15 @@ def unquote_string(quoted):
     Raises:
         TypeError: `quoted` was not a ``str``.
     """
-    tmp_quoted = quoted.strip()
```

Comment:
This was inconsistent with how other helper functions work, which is to require the caller to provide a pre-stripped string, therefore allowing the caller to skip this call (and creating of an additional string instance) when it is unnecessary.